applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp

```diff
@@ -501,14 +523,14 @@ inline void TreeContactSearch<TDim, TNumNodes>::InitializeAcceleration()
             auto it_node = nodes_array.begin() + i;
             const array_1d<double, 3>& nodal_volume_acceleration = it_node->FastGetSolutionStepValue(VOLUME_ACCELERATION);
             if (norm_2(volume_acceleration) > 0.0) {
-                if (!it_node->IsFixed(ACCELERATION_X)) it_node->FastGetSolutionStepValue(ACCELERATION_X) = volume_acceleration[0];
-                if (!it_node->IsFixed(ACCELERATION_Y)) it_node->FastGetSolutionStepValue(ACCELERATION_Y) = volume_acceleration[1];
-                if (!it_node->IsFixed(ACCELERATION_Z)) it_node->FastGetSolutionStepValue(ACCELERATION_Z) = volume_acceleration[2];
+                if (!it_node->IsFixed(ACCELERATION_X)) it_node->FastGetSolutionStepValue(ACCELERATION_X, 1) = volume_acceleration[0];
```

Comment:
this is inefficient since you access once to the array per each component.

    do something like 

    array_1d<double, 3>& old_acc = it_node->FastGetSolutionStepValue(ACCELERATION);
    if(...)
          old_acc[0] = ...
    if(...)
          old_acc[1] = ...

Reply:
This is removed in my branch, so Not need to modify (I will do a new PR)